src/Assertion/AssertionInterface.php

```diff
@@ -20,12 +20,11 @@ interface AssertionInterface
      * This method is passed the ACL, Role, Resource, and privilege to which the authorization query applies. If the
      * $role, $resource, or $privilege parameters are null, it means that the query applies to all Roles, Resources, or
      * privileges, respectively.
-     *
-     * @param  Acl                        $acl
-     * @param  RoleInterface         $role
-     * @param  ResourceInterface $resource
-     * @param  string                         $privilege
-     * @return bool
      */
-    public function assert(Acl $acl, RoleInterface $role = null, ResourceInterface $resource = null, $privilege = null);
+    public function assert(
+        Acl $acl,
+        ?RoleInterface $role = null,
+        ?ResourceInterface $resource = null,
+        ?string $privilege = null
+    ): bool;
```

Comment:
This is one of the different BC breaks. This would not work with existing applications and breaks with semantic versioning.